### PR TITLE
Adds mlx.core.radians and mlx.core.degrees and python bindings

### PR DESCRIPTION
**Pull request by aneeshk1412**
## Proposed changes

Adds `mlx.core.radians` and `mlx.core.degrees` and python bindings, which will complete issue (#1008).

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: python/tests/test_vmap.py [hunk 1]
```diff
@@ -49,8 +49,10 @@ def test_unary(self):
             "sin",
             "sqrt",
             "square",
+            "degrees",
+            "radians",
         ]
-        ops = ["erfinv"]
+        ops += ["erfinv"]
```
Comment (angeloskath) [inline python/tests/test_vmap.py:55]: Damn thanks for catching that. You could also do it as follows though
```suggestion
                "erfinv"
        ]
```

**Review by angeloskath — COMMENTED** [review]
This looks pretty straightforward and great to me, thanks!

Would you mind rebasing on main so that we can run the tests and then merge?

File: python/tests/test_vmap.py (hunk 1 above)
Comment (aneeshk1412) [reply python/tests/test_vmap.py]: Apparently `"erfinv"` is already there in the previous list, so I think we can just remove it from this line.

**Comment by aneeshk1412** [comment]
Rebased to current main branch.

**Review by angeloskath — APPROVED** [review]

**Pull request merged**